programs/util.h

```diff
@@ -84,7 +84,9 @@ extern "C" {
 /*-**************************************************************
 *  Basic Types
 *****************************************************************/
-#if  !defined (__VMS) && (defined (__cplusplus) || (defined (__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) /* C99 */) )
+#ifndef BASIC_TYPES_DEFINED
```

Comment:
I really prefer to keep the "mem.h" approach within `zstd`.
Unifying with LZ4 is a nice bonus, but a secondary one.

lz4 source has a different policy : the core library must remain "single file", with only `"lz4.c" + "lz4.h"`, and no other dependency required (except standard lib), which is obviously completely difference from zstd source tree.

Now, lz4 cli is not the "core library".
Here, it would be possible to bring in `mem.h` if it helps.